doc/man7/provider-mac.pod

```diff
@@ -108,7 +109,8 @@ I<mctx> parameter and return the duplicate copy.
 =head2 Encryption/Decryption Functions
 
 OSSL_FUNC_mac_init() initialises a mac operation given a newly created provider
-side mac context in the I<mctx> parameter.
+side mac context in the I<mctx> parameter.  The I<params> are set before setting
+the MAC I<key> of I<keylen> bytes.
```

Comment:
I wonder if there is any case that relies on the keylength being known before setting a param?

Reply:
Not yet :)

It's still possible to set params after the key.  It just can't be done using the `init()` call.  I don't think this is a problem.

Reply:
How do you know in what order the arguments are treated?  You know what *our* implementations do, but you have no idea what others will do.  Some might choose to treat *some* of the params before the key and keylength, and *some* after.  It really shouldn't matter, as long as they do something *sensible*, and that's up to each provider author to figure out.